api/v1alpha1/inferencepool_types.go

```diff
@@ -59,8 +59,86 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig configures the extension that runs the endpoint picking service.
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
+}
+
+// ExtensionConfig specifies how to configure an extension that runs the endpoint picker.
+type ExtensionConfig struct {
+	// ExtensionRef is a reference to a service extension.
+	ExtensionRef *ExtensionReference `json:"extensionRef"`
+
+	// ExtensionConnection configures the connection between the gateway and the extension.
+	ExtensionConnection `json:"extensionConnection"`
 }
 
+// ExtensionReference is a reference to the extension deployment.
+type ExtensionReference struct {
+	// Group is the group of the referent.
+	// When unspecified or empty string, core API group is inferred.
+	//
+	// +optional
+	// +kubebuilder:default=""
+	Group *string `json:"group,omitempty"`
+
+	// Kind is the Kubernetes resource kind of the referent. For example
+	// "Service".
+	//
+	// Defaults to "Service" when not specified.
+	//
+	// ExternalName services can refer to CNAME DNS records that may live
+	// outside of the cluster and as such are difficult to reason about in
+	// terms of conformance. They also may not be safe to forward to (see
+	// CVE-2021-25740 for more information). Implementations SHOULD NOT
+	// support ExternalName Services.
+	//
+	// Support: Core (Services with a type other than ExternalName)
+	//
+	// Support: Implementation-specific (Services with type ExternalName)
+	//
+	// +optional
+	// +kubebuilder:default=Service
+	Kind *string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	Name string `json:"name"`
+}
+
+// ExtensionConnection encapsulates options that configures the connection to the extension.
+type ExtensionConnection struct {
+	// The port number on the pods running the extension. Defaults to 9002 if not set.
+	//
+	// +kubebuilder:default=9002
+	TargetPortNumber *int32 `json:"targetPortNumber"`
```

Comment:
It's rather arbitrary, but I think we could make a compelling case for this being in the objectRef instead and without a default. I don't feel particularly strongly about this, but [backendRef in Gateway API includes the target port](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/object_reference_types.go#L146), so it would be a bit more consistent to mirror that pattern here. 


It's also possible that some kinds of backends will have a single port that would make this port number unnecessary. Since it's much easier to add a default in the future than to remove one, I'd recommend adjusting this so that implementations default to port 9002 when Service is the target and no port has been specified, instead of defaulting in the API.

Reply:
Sounds good, I had it there initially, but moved it here later.